uni/lib/controller/fetchers/schedule_fetcher/schedule_fetcher_new_api.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:uni/controller/fetchers/schedule_fetcher/schedule_fetcher.dart';
+import 'package:uni/controller/networking/network_router.dart';
+import 'package:uni/controller/parsers/schedule/new_api/parser.dart';
+import 'package:uni/model/entities/lecture.dart';
+import 'package:uni/model/entities/profile.dart';
+import 'package:uni/model/entities/session.dart';
+
+/// Class for fetching the user's lectures from the schedule's HTML page.
+class ScheduleFetcherNewApi extends ScheduleFetcher {
+  @override
+  List<String> getEndpoints(Session session) {
+    final urls = NetworkRouter.getBaseUrlsFromSession(session)
+        .map((url) => '${url}hor_geral.estudantes_view')
+        .toList();
+    return urls;
+  }
+
+  /// Fetches the user's lectures from the schedule's HTML page.
+  @override
+  Future<List<Lecture>> getLectures(Session session, Profile profile) async {
+    final endpoints = getEndpoints(session);
+    final lectiveYear = getLectiveYear(DateTime.now());
+
+    final futures = <Future<List<Lecture>>>[];
```

Comment:
Opportunity to a more functional programming here with `.map`

Reply:
Sounds good! I'll do that.